gtda/diagrams/_metrics.py

```diff
@@ -49,8 +54,8 @@ def heats(diagrams, sampling, step_size, sigma):
     diagrams = np.array((diagrams - sampling[0]) / step_size,
                         dtype=int)
 
-    [_heat(heats_[i], sampled_diag, sigma)
-        for i, sampled_diag in enumerate(diagrams)]
+    heats_ = [_heat(heats_[i], sampled_diag, sigma)
```

Comment:
Je pense que c'est plus rapide rester inplace plutot que outofplace

Reply:
peut-etre, meme gaussian_filter n'est pas in-place. Donc, heats_[i] ne subit pas de smoothing.